src/test/run-fail/explicit-panic-msg.rs

```diff
@@ -15,5 +15,6 @@
 fn main() {
     let mut a = 1;
     if 1 == 1 { a = 2; }
-    panic!(format!("woooo{}", "o"));
+    let msg = format!("woooo{}", "o");
+    panic!(msg);
```

Comment:
on reflection, the benefit of the warning (catching the programming mistake where one passes a format literal string to `panic!` but forgets the associated format arguments) does not warrant breaking even code like the example above.

I still think the fact that my macro did not work represents a bug somewhere, perhaps in the macro system or in the stability checker.

But for now I am closing this PR; we can put in this warning later (i.e. after 1.0).